test/FrequencyMapAgentTest.cpp

```diff
@@ -247,10 +348,108 @@ TEST_F(FrequencyMapAgentTest, adjust_platform_uncore)
     }
 }
 
-TEST_F(FrequencyMapAgentTest, split_policy)
+TEST_F(FrequencyMapAgentTest, adjust_platform_gpu)
+{
+    setup_gpu(M_NUM_GPU);
+    std::vector<double> policy(m_num_policy, NAN);
+    set_expectations_adjust_platform_init(M_NUM_GPU);
+    m_agent->adjust_platform(policy);
+    // initial all-NAN policy is accepted
+    EXPECT_FALSE(m_agent->do_write_batch());
+
+    policy[DEFAULT] = 1.3e9;
+    policy[GPU_DEFAULT] = 1.1e9;
+    EXPECT_CALL(*m_platform_io, adjust(FREQ_CONTROL_IDX, policy[DEFAULT]))
+        .Times(M_NUM_CPU);
+    EXPECT_CALL(*m_platform_io, adjust(GPU_FREQ_MIN_CONTROL_IDX, policy[GPU_DEFAULT]))
+        .Times(M_NUM_GPU);
+    EXPECT_CALL(*m_platform_io, adjust(GPU_FREQ_MAX_CONTROL_IDX, policy[GPU_DEFAULT]))
+        .Times(M_NUM_GPU);
+    m_agent->adjust_platform(policy);
+    EXPECT_TRUE(m_agent->do_write_batch());
+
+    //Change GPU frequency
+    policy[GPU_DEFAULT] = 1.4e9;
+    EXPECT_CALL(*m_platform_io, adjust(GPU_FREQ_MIN_CONTROL_IDX, policy[GPU_DEFAULT]))
+        .Times(M_NUM_GPU);
+    EXPECT_CALL(*m_platform_io, adjust(GPU_FREQ_MAX_CONTROL_IDX, policy[GPU_DEFAULT]))
+        .Times(M_NUM_GPU);
+    m_agent->adjust_platform(policy);
+    EXPECT_TRUE(m_agent->do_write_batch());
+
+    //Don't write if the GPU frequency is unchanged
+    policy[GPU_DEFAULT] = 1.4e9;
+    m_agent->adjust_platform(policy);
+    EXPECT_FALSE(m_agent->do_write_batch());
+
+    //Don't write if GPU frequency changes to NAN
+    policy[GPU_DEFAULT] = NAN;
+    m_agent->adjust_platform(policy);
+    EXPECT_FALSE(m_agent->do_write_batch());
+}
+
+TEST_F(FrequencyMapAgentTest, adjust_platform_nogpu)
 {
+    int num_gpu = 0;
+    setup_gpu(num_gpu);
+
+    std::vector<double> policy(m_num_policy, NAN);
+    set_expectations_adjust_platform_init(num_gpu);
+    m_agent->adjust_platform(policy);
+    // initial all-NAN policy is accepted
+    EXPECT_FALSE(m_agent->do_write_batch());
+
+    policy[DEFAULT] = m_freq_min;
+    policy[UNCORE] = 1.8e9;
+    policy[GPU_DEFAULT] = 1.1e9;
+
+    EXPECT_CALL(*m_platform_io, adjust(FREQ_CONTROL_IDX, m_freq_min))
+        .Times(M_NUM_CPU);
+    EXPECT_CALL(*m_platform_io, adjust(GPU_FREQ_MIN_CONTROL_IDX, policy[GPU_DEFAULT]))
+        .Times(num_gpu);
+    EXPECT_CALL(*m_platform_io, adjust(GPU_FREQ_MAX_CONTROL_IDX, policy[GPU_DEFAULT]))
+        .Times(num_gpu);
+    EXPECT_CALL(*m_platform_io, adjust(UNCORE_MIN_CTL_IDX, 1.8e9));
+    EXPECT_CALL(*m_platform_io, adjust(UNCORE_MAX_CTL_IDX, 1.8e9));
+    m_agent->adjust_platform(policy);
+    EXPECT_TRUE(m_agent->do_write_batch());
+
+    //Changing GPU frequency only should result in no changes
+    policy[GPU_DEFAULT] = 1.4e9;
+    m_agent->adjust_platform(policy);
+    EXPECT_CALL(*m_platform_io, adjust(GPU_FREQ_MIN_CONTROL_IDX, policy[GPU_DEFAULT]))
+        .Times(num_gpu);
+    EXPECT_CALL(*m_platform_io, adjust(GPU_FREQ_MAX_CONTROL_IDX, policy[GPU_DEFAULT]))
+        .Times(num_gpu);
+    EXPECT_FALSE(m_agent->do_write_batch());
```

Comment:
This is going to have to change based on the warning we're promoting to an error.